internal/etcd/etcd.go

```diff
@@ -227,25 +228,7 @@ func (c *ClientEtcdAPI) Defragment(ctx context.Context, member Member) error {
 }
 
 func checkVersion(config Config) (*Versions, error) {
-	tr := &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		Dial: (&net.Dialer{
-			Timeout:   2 * time.Second,
-			KeepAlive: 2 * time.Second,
-		}).Dial,
-		TLSHandshakeTimeout:   2 * time.Second,
-		IdleConnTimeout:       2 * time.Second,
-		ResponseHeaderTimeout: 2 * time.Second,
-	}
-
-	if config.TLS != nil {
-		tr.TLSClientConfig = config.TLS
-	}
-
-	httpClient := &http.Client{
-		Transport: tr,
-		Timeout:   2 * time.Second,
-	}
+	httpClient := createHTTPClient(config)
```

Comment:
Is the name of the function still represents behavior?

Reply:
Yeah, `checkVersion` still checks the version and `createHTTPClient` creates a http client